src/safeds_runner/server/memoization_map.py

```diff
@@ -0,0 +1,177 @@
+        time_last_access = time.time_ns()
+        time_compute = time_compute_end - time_compute_start
+        value_memory = _get_size_of_value(result)
+        self.map_values[key] = result
+        memoization_stats = MemoizationStats(time_last_access, time_compute, time_compare, value_memory)
+        logging.info(f"New memoization stats for {function_name}: {memoization_stats}")
+        self.map_stats[key] = memoization_stats
+        return result
+
+
+def _convert_list_to_tuple(values: list) -> tuple:
+    """
+    Recursively convert a mutable list of values to an immutable tuple containing the same values, to make the values hashable.
+
+    Parameters
+    ----------
+    values : list
+        Values that should be converted to a tuple
+
+    Returns
+    -------
+    tuple
+        Converted list containing all the elements of the provided list
+    """
+    return tuple(_convert_list_to_tuple(value) if isinstance(value, list) else value for value in values)
+
+
+def _get_size_of_value(value: Any) -> int:
+    """
+    Recursively calculate the memory usage of a given value.
+
+    Parameters
+    ----------
+    value
+        Any value of which the memory usage should be calculated.
+
+    Returns
+    -------
+    Size of the provided value in bytes
+    """
+    size_immediate = sys.getsizeof(value)
+    if isinstance(value, dict):
+        return sum(map(_get_size_of_value, value.items())) + size_immediate
+    elif isinstance(value, list) or isinstance(value, tuple) or isinstance(value, set) or isinstance(value, frozenset):
+        return sum(map(_get_size_of_value, value)) + size_immediate
+    elif isinstance(value, Table):
+        return _get_size_of_value(value._data) + _get_size_of_value(value._schema) + size_immediate
+    elif isinstance(value, Schema):
+        return _get_size_of_value(value._schema) + size_immediate
+    elif isinstance(value, Image):
+        return _get_size_of_value(
+            value._image_tensor) + value._image_tensor.element_size() * value._image_tensor.nelement() + size_immediate
+    elif isinstance(value, Column):
+        return _get_size_of_value(value._data) + _get_size_of_value(value._name) + _get_size_of_value(
+            value._type) + size_immediate
+    elif isinstance(value, Row):
+        return _get_size_of_value(value._data) + _get_size_of_value(value._schema) + size_immediate
+    elif isinstance(value, TaggedTable):
+        return _get_size_of_value(value._data) + _get_size_of_value(value._schema) + _get_size_of_value(
+            value._features) + _get_size_of_value(value._target) + size_immediate
+    # elif isinstance(value, TimeSeries):
+    #     return _get_size_of_value(value._data) + _get_size_of_value(value._schema) + _get_size_of_value(
+    #         value._time) + _get_size_of_value(value._features) + _get_size_of_value(value._target) + size_immediate
```

Comment:
@lars-reimann It would be more maintainable if this was part of the library. If this is considered useful, I would PR it to the library. The relevant classes would be made compatible with the other built-in types by overriding the `__sizeof__` method and returning the appropriate result.

Reply:
Sure, that seems useful.

Reply:
Could another (minor) release of the library be prepared, if it doesn't take too long?
This would (more easily) unblock this PR

Reply:
No problem, the release action is running right now.